pkg/config/config.go

```diff
@@ -111,6 +119,29 @@ func LoadFile(configPath string, relativePath ...string) (Config, error) {
 	return config, nil
 }
 
+// getEmbeddedConfig returns the embedded configuration based on the provided network mode.
+func getEmbeddedConfig(configName string) (Config, error) {
+	var cfgData []byte
+
+	switch configName {
+	case netmode.MainNet.String():
+		cfgData = config.Mainnet
+	case netmode.TestNet.String():
+		cfgData = config.Testnet
+	case netmode.PrivNet.String():
+		cfgData = config.Privnet
+	default:
+		return Config{}, fmt.Errorf("unsupported network mode: %s", configName)
+	}
+
+	var cfg Config
+	err := yaml.Unmarshal(cfgData, &cfg)
+	if err != nil {
+		return cfg, fmt.Errorf("failed to unmarshal embedded config: %w", err)
+	}
+	return cfg, nil
```

Comment:
Do not unmarshal here, there's already a spacial unmarshalling code inside the `LoadFile`.